### PR TITLE
fix: Remove zstd from API

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
**Changes:**
- Remove ZSTD configuration

### CODE REVIEW

File: internal/pkg/service/common/configmap/flags.go [hunk 1]
```diff
@@ -112,10 +112,10 @@ func mapAndFilterField() OnField {
 		// Field must have tag
 		if tag, found := field.Tag.Lookup(configKeyTag); found {
 			parts := strings.Split(tag, tagValuesSeparator)
-			if fieldName = parts[0]; fieldName == "" && len(parts) == 2 && parts[1] == "squash" {
+			if fieldName = parts[0]; fieldName == "-" && len(parts) == 2 && parts[1] == "squash" {
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/common/configmap/flags.go:115]: :+1: Maybe both options should be supported (`<empty>` and `-`), ideally if it works the same as `json`/`yaml` tags.

[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/common/configmap/flags.go:115]: :thinking: 

I think, you should return `"", FALSE` for `<empty>` and `-`.

https://github.com/keboola/keboola-as-code/blob/7f335bc2a4fd82454b78b85a2b50e796d742984b/internal/pkg/service/common/configmap/flags.go#L110-L126

-----------------

Now you don't ignore it, but take it as `squashed'`.

If the type is not struct, the result is same, but it is not correct.

-------------------

Here is test for it:

https://github.com/keboola/keboola-as-code/blob/7f335bc2a4fd82454b78b85a2b50e796d742984b/internal/pkg/service/common/configmap/configmap_test.go#L47-L49

--------------------

Read more: https://pkg.go.dev/github.com/mitchellh/mapstructure#hdr-Embedded_Structs_and_Squashing

**Comment by michaljurecko** [comment]
Side note: In the tests, it is better to skip the validation of the structure for ZSTD values. It's better than commenting on them completely.

File: internal/pkg/service/stream/storage/level/local/diskreader/reader_test.go [hunk 2]
```diff
@@ -103,17 +103,17 @@ func TestVolume_NewReaderFor_Compression(t *testing.T) {
 			LocalCompression:   compression.NewNoneConfig(),
 			StagingCompression: compression.NewGZIPConfig(),
 		},
-		{
+		/*{
 			Name:               "None_To_ZSTD",
 			LocalCompression:   compression.NewNoneConfig(),
 			StagingCompression: compression.NewZSTDConfig(),
-		},
+		},*/
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/level/local/diskreader/reader_test.go:110]: https://github.com/keboola/keboola-as-code/pull/1901#issuecomment-2220655646

File: internal/pkg/service/common/configmap/flags.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/common/configmap/flags.go:115]: Similar code in `configpatch` package, logic must match:

```
https://github.com/keboola/keboola-as-code/blob/7f335bc2a4fd82454b78b85a2b50e796d742984b/internal/pkg/service/common/configpatch/visit.go#L180-L191
```

[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/common/configmap/flags.go:115]: I think, problem is here, `json:"zstd,...`:
image

--------------

Options:
1. `json:""`
2. Modify logic of the `configpatch` pkg, take into account the value in the Config structure.

File: internal/pkg/service/stream/storage/level/local/encoding/compression/config_test.go [hunk 3]
```diff
@@ -128,7 +128,7 @@ func TestConfig_Validation(t *testing.T) {
 				},
 			},
 		},
-		{
+		/*{
```
Comment (Matovidlo) [inline internal/pkg/service/stream/storage/level/local/encoding/compression/config_test.go:131]: This should be commented?

**Review by Matovidlo — APPROVED** [review]
LGTM, just commented code

File: internal/pkg/service/common/configmap/flags.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/common/configmap/flags.go]: Looks fine for me.

[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/common/configmap/flags.go]: Yes, thanks.

**Review by michaljurecko — APPROVED** [review]

**Pull request merged**